src/libstd/sync/mutex.rs

```diff
@@ -393,6 +393,19 @@ impl<T: ?Sized + fmt::Debug> fmt::Debug for Mutex<T> {
     }
 }
 
+#[stable(feature = "std_guard_impls", since = "1.20")]
+impl<T: ?Sized + fmt::Display> fmt::Display for Mutex< T> {
```

Comment:
Without `Mutex` being `Display`, it's a lot more difficult for `MutexGuard` to be `Display` as `self.__lock` can't be `Display`ed